mail/postfix/Makefile

```diff
@@ -177,136 +181,78 @@ newaliases_path=/usr/bin/newaliases
-	$(INSTALL_DIR) $(1)$(mail_spool_directory)
 	$(INSTALL_DIR) $(1)/etc/init.d/
 	$(INSTALL_BIN) ./files/postfix.init $(1)/etc/init.d/postfix
 endef
 
 define Package/postfix/postinst
 #!/bin/sh
 
- if [ -f "$${IPKG_INSTROOT}$(sendmail_path)" -a "$$(readlink "$${IPKG_INSTROOT}$(sendmail_path)")" != "$(sendmail_path)$(ln_suffix)" ]; then
-  mv "$${IPKG_INSTROOT}$(sendmail_path)" "$${IPKG_INSTROOT}$(sendmail_path)$(ln_old_suffix)"
-  echo "Warning: $${IPKG_INSTROOT}$(sendmail_path) saved as $${IPKG_INSTROOT}$(sendmail_path)$(ln_old_suffix)"
- fi
- if [ ! -f "$${IPKG_INSTROOT}$(sendmail_path)" ]; then
-  ln -s "$(sendmail_path)$(ln_suffix)" "$${IPKG_INSTROOT}$(sendmail_path)"
- fi
-
- if [ -f "$${IPKG_INSTROOT}$(newaliases_path)" -a "$$(readlink "$${IPKG_INSTROOT}$(newaliases_path)")" != "$(newaliases_path)$(ln_suffix)" ]; then
-  mv "$${IPKG_INSTROOT}$(newaliases_path)" "$${IPKG_INSTROOT}$(newaliases_path)$(ln_old_suffix)"
-  echo "Warning: $${IPKG_INSTROOT}$(newaliases_path) saved as $${IPKG_INSTROOT}$(newaliases_path)$(ln_old_suffix)"
- fi
- if [ ! -f "$${IPKG_INSTROOT}$(newaliases_path)" ]; then
-  ln -s "$(newaliases_path)$(ln_suffix)" "$${IPKG_INSTROOT}$(newaliases_path)"
- fi
-
- if [ -f "$${IPKG_INSTROOT}$(mailq_path)" -a "$$(readlink "$${IPKG_INSTROOT}$(mailq_path)")" != "$(mailq_path)$(ln_suffix)" ]; then
-  mv "$${IPKG_INSTROOT}$(mailq_path)" "$${IPKG_INSTROOT}$(mailq_path)$(ln_old_suffix)"
-  echo "Warning: $${IPKG_INSTROOT}$(mailq_path) saved as $${IPKG_INSTROOT}$(mailq_path)$(ln_old_suffix)"
- fi
- if [ ! -f "$(mailq_path)" ]; then
-  ln -s "$(mailq_path)$(ln_suffix)" "$${IPKG_INSTROOT}$(mailq_path)"
- fi
-
- grep -qc main\.cf "$${IPKG_INSTROOT}"/etc/sysupgrade.conf >/dev/null || echo "$(config_directory)/main.cf" >> "$${IPKG_INSTROOT}"/etc/sysupgrade.conf
- grep -qc master\.cf "$${IPKG_INSTROOT}"/etc/sysupgrade.conf >/dev/null || echo "$(config_directory)/master.cf" >> "$${IPKG_INSTROOT}"/etc/sysupgrade.conf
- grep -qc aliases "$${IPKG_INSTROOT}"/etc/sysupgrade.conf >/dev/null || echo "$(config_directory)/aliases" >> "$${IPKG_INSTROOT}"/etc/sysupgrade.conf
- grep -qc virtual "$${IPKG_INSTROOT}"/etc/sysupgrade.conf >/dev/null || echo "$(config_directory)/virtual" >> "$${IPKG_INSTROOT}"/etc/sysupgrade.conf
-
- touch "$${IPKG_INSTROOT}$(config_directory)"/opkg_postinst
-
- if [ -z "$${IPKG_INSTROOT}" ]; then
-  ps | grep "postfix/master" | grep -cvq grep >/dev/null && /etc/init.d/postfix restart
```

Comment:
When I install Postfix from the official packages repo, I get this:
```
root@LEDE:~# opkg install ./postfix_3.2.3-1_arm_cortex-a9.ipk 
Installing postfix (3.2.3-1) to root...
Configuring postfix.
postfix/postfix-script: starting the Postfix mail system
Collected errors:
 * pkg_run_script: package "postfix" postinst script returned status 1.
 * opkg_configure: postfix.postinst returned 1.
root@LEDE:~# opkg --autoremove remove postfix
Removing package postfix from root...
postfix/postfix-script: stopping the Postfix mail system
postfix/postfix-script: fatal: the Postfix mail system is not running
Not deleting modified conffile /etc/postfix/main.cf.
root@LEDE:~# 
```
The error during installation is caused by /etc/init.d/postfix restart. The default postinst script executes "/etc/init.d/postfix enable" and "/etc/init.d/postfix start", see lines 250-260 of [functions.sh](https://git.lede-project.org/?p=source.git;a=blob;f=package/base-files/files/lib/functions.sh;h=81ef84b8ef8fc44ae8731b80766c222094408010;hb=HEAD#l250). If Postfix is already started by the package-supplied postinst, functions.sh generates an error.

I should also note that "postdrop" user is not created at present when Postfix is installed from the official packages repo. Somehow "postfix check" does not detect this deficiency.

Lastly, "opkg upgrade" involves running prerm and postinst scripts: see line 1462 in [opkg_install.c](https://git.lede-project.org/?p=project/opkg-lede.git;a=blob;f=libopkg/opkg_install.c;h=e6f8a1b6276ede518a5c59b2f9347f1de8e5dd7a;hb=HEAD#l1462), lines 26-78 in [opkg_upgrade.c](https://git.lede-project.org/?p=project/opkg-lede.git;a=blob;f=libopkg/opkg_upgrade.c;h=211267410bc6748560b0e10f2dd7eb57ca2bfa01;hb=HEAD#l26) and line 548 in [opkg_install.c](https://git.lede-project.org/?p=project/opkg-lede.git;a=blob;f=libopkg/opkg_install.c;h=e6f8a1b6276ede518a5c59b2f9347f1de8e5dd7a;hb=HEAD#l548). It is therefore unnecessary to restart Postfix on upgrade since opkg and /lib/functions.sh do it automatically on every upgrade.

Reply:
Ok, I agree.

Reply:
Yes, I have tested fresh installs, upgrades and uninstalls from fresh installs and upgrades. I am no longer getting installation, upgrade or uninstall errors.